src/omv/ports/stm32/sensor.c

```diff
@@ -229,13 +228,17 @@ int sensor_dcmi_config(uint32_t pixformat) {
     return 0;
 }
 
-int sensor_abort() {
+int sensor_abort(bool fifo_flush, bool in_irq) {
     // This stops the DCMI hardware from generating DMA requests immediately and then stops the DMA
     // hardware. Note that HAL_DMA_Abort is a blocking operation. Do not use this in an interrupt.
     if (DCMI->CR & DCMI_CR_ENABLE) {
         DCMI->CR &= ~DCMI_CR_ENABLE;
-        HAL_DMA_Abort(&DMAHandle);
-        HAL_NVIC_EnableIRQ(DMA2_Stream1_IRQn);
+        if (in_irq) {
+            HAL_DMA_Abort_IT(&DMAHandle);
+        } else {
+            HAL_DMA_Abort(&DMAHandle);
+        }
+        HAL_NVIC_DisableIRQ(DMA2_Stream1_IRQn);
         #if defined(OMV_MDMA_CHANNEL_DCMI_0)
         HAL_MDMA_Abort(&DCMI_MDMA_Handle0);
         HAL_MDMA_Abort(&DCMI_MDMA_Handle1);
```

Comment:
```suggestion
        if (!in_irq) {
            HAL_MDMA_Abort(&DCMI_MDMA_Handle0);
            HAL_MDMA_Abort(&DCMI_MDMA_Handle1);
        }
```

Reply:
Looks good otherwise.

Reply:
That doesn't seem to have any effect either. I think it's safe to abort it unconditionally, but I can make that change if it's necessary.

Reply:
The aborts technically have an infinite loop in them. So, you want to not call them in an IRQ.

Reply:
Good point, fixed.